### DISCUSSION TITLE
Proper way to do Edge perturbation for node classification

### DISCUSSION THREAD

**Discussion by yorgancioglukaan** [pyg-team/pytorch_geometric#8938] [Q&A] [answered]
Hi all,

I want to perform an edge perturbation alaysis where I keep removing an increasing portion of edges before perfroming node classification. (i.e delete 1/2 of edges from original graph, then 3/4, 7/8,...). I have 2 questions about this. 

1. Is modifying [RandomLinkSplit](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.transforms.RandomLinkSplit.html#torch-geometric-transforms-randomlinksplit) a good place to start, or is there a better way to do this? Keep in mind these splits will be completely seperate from train/val/test sets for the nodes.  

3. I am concerned that with very high sparseness there will be unconnected components or even 0-degree nodes. How well does PyG handle this? Should I construct a minimum spanning tree and add those edges to the dataset just in case?

I would appreciate any guidance. Thanks!

**Comment by rusty1s** [comment, marked as answer]
I guess you want to use `utils.dropout_edge` for this. If this results in unconnected components/isolated nodes, then this is generally not a problem (except for that a node could only learn from itself or its component).

**Reply by yorgancioglukaan** [reply]
That makes sense. Thanks!
